tests/Tasks/CollectionFactoryTest.php

```diff
@@ -89,36 +89,43 @@ public function testRunTask()
             ],
             'options' => [
                 'filepath' => $filePath,
+                'array-opt' => [
+                    'opt1',
+                    'opt2',
+                ],
             ],
         ];
         $this->taskCollectionFactory($tasks)->run();
-        $this->assertSame(__METHOD__, file_get_contents($filePath));
+        $this->assertSame(__METHOD__ . 'opt1opt2', file_get_contents($filePath));
     }
 
     /**
      * Tests the 'exec' task.
      */
     public function testExecTask(): void
     {
+        $filePath = $this->getSandboxFilepath('test-file-exec.txt');
+
         $tasks = [
             [
                 'task' => 'exec',
-                'command' => 'touch',
+                'command' => __DIR__ . '/../../bin/run',
                 'arguments' => [
-                    'file.txt',
+                    'custom:test',
+                    __METHOD__,
                 ],
                 'options' => [
-                    // 1980-06-06 23:59:59.
-                    '-t' => '198006062359.59'
+                    'filepath' => $filePath,
+                    'array-opt' => [
+                        'opt1',
+                        'opt2',
+                    ],
```

Comment:
This example is confusing, it seems like an array of options is passed, but actually this is an option that has a value of type array.

Reply:
In console this is equivalent to `--array-opt=opt1 --array-opt=opt2`

Reply:
Yes, I can see this from the implementation, but it is not clear, it could also be interpreted as a difference between value options and flag options. You are using the word `array-opt` for what is the `option-name` and `opt{1|2}` for the values.

I plan to replace it with this which is a much clearer example of how this is intended to be used:

```
'recipients' => [
  'jack@example.com',
  'jill@example.com',
]
``